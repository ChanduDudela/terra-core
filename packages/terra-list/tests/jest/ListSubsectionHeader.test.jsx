import React from 'react';
import ListSubsectionHeader from '../../src/ListSubsectionHeader';

// Snapshot Tests
it('should render default', () => {
  const shallowComponent = shallow(<ListSubsectionHeader />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render with title', () => {
  const shallowComponent = shallow(<ListSubsectionHeader title="Test Title" />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render with title and level', () => {
  const shallowComponent = shallow(<ListSubsectionHeader title="Test Title" level={3} />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render with isCollapsed', () => {
  const shallowComponent = shallow(<ListSubsectionHeader isCollapsed />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render with isCollapsible', () => {
  const shallowComponent = shallow(<ListSubsectionHeader isCollapsible />);
  expect(shallowComponent).toMatchSnapshot();
});

it('should render with callback functions', () => {
  const shallowComponent = shallow(<ListSubsectionHeader refCallback={jest.fn()} onSelect={jest.fn()} />);
  expect(shallowComponent).toMatchSnapshot();
});
