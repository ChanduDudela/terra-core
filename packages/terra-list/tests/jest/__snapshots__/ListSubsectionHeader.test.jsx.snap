// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default 1`] = `
<li
  className="subsection-header"
>
  <h2
    className="title"
  />
</li>
`;

exports[`should render with callback functions 1`] = `
<li
  className="subsection-header"
>
  <h2
    className="title"
  />
</li>
`;

exports[`should render with isCollapsed 1`] = `
<li
  className="subsection-header"
>
  <h2
    className="title"
  />
</li>
`;

exports[`should render with isCollapsible 1`] = `
<li
  aria-expanded={true}
  aria-level={2}
  className="subsection-header is-collapsible"
  data-item-show-focus="true"
  onBlur={[Function]}
  onMouseDown={[Function]}
  role="heading"
  tabIndex="0"
>
  <div
    className="start"
  >
    <span
      className="accordion-icon is-open"
    />
  </div>
  <div
    className="fill"
  >
    <h2
      className="title"
    />
  </div>
</li>
`;

exports[`should render with title 1`] = `
<li
  className="subsection-header"
>
  <h2
    className="title"
  >
    Test Title
  </h2>
</li>
`;

exports[`should render with title and level 1`] = `
<li
  className="subsection-header"
>
  <h3
    className="title"
  >
    Test Title
  </h3>
</li>
`;
